src/repositories/repoBranchesRepository.ts

```diff
@@ -0,0 +1,22 @@
+import { Db } from 'mongodb';
+import { BaseRepository } from "./baseRepository";
+import { Job } from "../entities/job"
+import { ILogger } from "../services/logger";
+import { IConfig } from 'config';
+
+export class RepoBranchesRepository extends BaseRepository<Job> {
+
+    constructor(db: Db, config: IConfig, logger: ILogger) {
+        super(config, logger, "RepoEntitlementsRepository", db.collection(config.get("repoBranchesCollection")));
+    }
+
+    async getRepoBranchesByRepoName(repoName: string): Promise<any> {
+        const query = { 'repoName': repoName };
+        const repoDetails = await this.findOne(query, `Mongo Timeout Error: Timedout while retrieving Repoinformation for ${repoName}`);
+        if (repoDetails && repoDetails.bucket && repoDetails.url) {
+            return repoDetails
+        } else {
+            return { status: 'failure' };
```

Comment:
We might want to make this a little more robust with error messaging - using `this._logger` would be useful here, especially to message to a writer that no results were found for the repo